Sources/SwiftFormat/Rules/DontRepeatTypeInStaticProperties.swift

```diff
@@ -110,3 +67,40 @@ extension Finding.Message {
     "remove the suffix '\(type)' from the name of the variable '\(name)'"
   }
 }
+
+extension Syntax {
+  /// Returns the name of the immediately enclosing type of this node if there is one,
+  /// otherwise nil.
+  fileprivate var typeName: String? {
```

Comment:
The name of this property feels a little misleading; when I saw you accessing the `typeName` of a `VariableDeclSyntax` node above, it read to me like it was accessing the type of the *variable*. Can we call it `containingDeclName` so that the nature of the traversal is clear at the call site?